netx/errorx/errorx.go

```diff
@@ -156,16 +163,23 @@ type SafeErrWrapperBuilder struct {
 
 	// TransactionID is the transaction ID, if any
 	TransactionID int64
+
+	// QuicErr indicates that the QUIC transport is used
+	QuicErr bool
 }
 
 // MaybeBuild builds a new ErrWrapper, if b.Error is not nil, and returns
 // a nil error value, instead, if b.Error is nil.
 func (b SafeErrWrapperBuilder) MaybeBuild() (err error) {
+	failureString := toFailureString
+	if b.QuicErr {
+		failureString = toQUICFailureString
+	}
```

Comment:
We should chat about this part of the design. I would have instinctively put checks for QUIC related errors into the function we're already using for checking for errors. Maybe I'm missing something?

Reply:
Yes, this would probably be more clear. In doing that, should, for example, a timeout error, induce the same notation in the result file, regardless whether it happened during a QUIC or a TCP connection? Or do you prefer having a `generic_timeout_error` and a `quic_generic_timeout_error`?

Reply:
I would keep a `generic_timeout_error` for now. I think maybe in the future we'll need to have more specific or different errors. But, for now, it seems to me we understand what is going on by combining operation and error.

(The error is `generic_timeout_error` rather than just `timeout` or `timeout_error` for legacy reasons that I honestly do not remember. Every time I write a table in LaTeX, I regret that this name is so long.)

Reply:
So I moved the QUIC specific content from the `toQUICFailureString` function into the `toFailureString` function and updated the tests. I added the QUIC specific failures in a batch at the end of the function. I'm not sure if the order should instead be based on the failure type.